classes/tests/test_events.py

```diff
@@ -14,50 +16,75 @@
 class EventsTests(unittest.TestCase):
     def setUp(self) -> None:
         self.world = World(40, initial_location_depot=False)
+        self.world.stack = []
+        self.vehicle = self.world.state.vehicles[0]
+        self.vehicle.current_location = self.world.state.clusters[0]
         self.large_world = World(
             40, sample_size=500, number_of_clusters=20, initial_location_depot=False
         )
-        self.world.state.current_location = self.world.state.clusters[0]
-        self.large_world.state.current_location = self.large_world.state.clusters[0]
+        self.large_world.stack = []
+        self.vehicle_large_world = self.large_world.state.vehicles[0]
+        self.vehicle_large_world.current_location = self.large_world.state.clusters[0]
+
         self.departure_time = 1
         self.travel_time = 5
 
     def test_scooter_departure(self):
         departure_event = ScooterDeparture(
-            self.departure_time, self.world.state.current_location.id
+            self.departure_time, self.vehicle.current_location.id
         )
-
+        # Remove all scooters from the scooter departure location
+        self.vehicle.current_location.scooters = []
         departure_event.perform(self.world)
 
         # test if the time of world object is set to the departure time
         self.assertEqual(departure_event.time, self.world.time)
 
-        arrival_event = next(iter(self.world.stack))
+        # Check that a lost trip event is created
+        self.assertIsInstance(self.world.stack.pop(), LostTrip)
+
+        # create new departure event with scooters in departure cluster
+        new_destination = random.choice(
+            [
+                cluster
+                for cluster in self.world.state.clusters
+                if len(cluster.scooters) > 0
+                and cluster.id != self.vehicle.current_location.id
+            ]
+        )
+        departure_event = ScooterDeparture(11, new_destination.id)
+        departure_event.perform(self.world)
+
+        arrival_event = self.world.stack.pop()
+
+        # Check that a lost trip event is created
+        self.assertIsInstance(arrival_event, ScooterArrival)
```

Comment:
I think it has gone a bit fast in the turns here, the comment doesn't explain the code underneath.